bcipy/signal/model/mach_learning/demo/demo_density_estimation.py

```diff
@@ -21,35 +20,30 @@ def _demo_kde():
     x_plot = np.linspace(-5, 10, 1000)[:, np.newaxis]
 
     # generate a dummy density function to sample data from
-    true_dens = (0.3 * norm(0, 1).pdf(x_plot[:, 0])
-                 + 0.7 * norm(5, 1).pdf(x_plot[:, 0]))
+    true_dens = 0.3 * norm(0, 1).pdf(x_plot[:, 0]) + 0.7 * norm(5, 1).pdf(x_plot[:, 0])
 
     fig, ax = plt.subplots()
-    ax.fill(x_plot[:, 0], true_dens, fc='black', alpha=0.2,
-            label='input distribution')
+    ax.fill(x_plot[:, 0], true_dens, fc="black", alpha=0.2, label="input distribution")
 
     # thumb up rule for bandwidth selection
-    bandwidth = 1.06 * min(
-        np.std(x), iqr(x) / 1.34) * np.power(x.shape[0], -0.2)
+    bandwidth = 1.06 * min(np.std(x), iqr(x) / 1.34) * np.power(x.shape[0], -0.2)
 
     # try different kernels and show how the look like
-    for kernel in ['gaussian', 'tophat', 'epanechnikov']:
-        kde = KernelDensityEstimate(kernel=kernel, bandwidth=bandwidth,
-                                    num_cls=1)
+    for kernel in ["gaussian", "tophat", "epanechnikov"]:
+        kde = KernelDensityEstimate(kernel=kernel, bandwidth=bandwidth, num_cls=1)
         kde.fit(x, y)
         log_dens = kde.list_den_est[0].score_samples(x_plot)
-        ax.plot(x_plot[:, 0], np.exp(log_dens), '-',
-                label="kernel = '{0}'".format(kernel))
+        ax.plot(x_plot[:, 0], np.exp(log_dens), "-", label="kernel = '{0}'".format(kernel))
 
     ax.text(6, 0.38, "N={0} points".format(n))
 
-    ax.legend(loc='upper left')
-    ax.plot(x[:, 0], -0.005 - 0.01 * np.random.random(x.shape[0]), '+k')
+    ax.legend(loc="upper left")
+    ax.plot(x[:, 0], -0.005 - 0.01 * np.random.random(x.shape[0]), "+k")
 
     ax.set_xlim(-4, 9)
     ax.set_ylim(-0.02, 0.4)
     plt.show()
-    print('KDE Flows!')
+    print("KDE Flows!")
     return 0
```

Comment:
Why return 0? If no return is expected, it can be left off and None will be returned.

Reply:
These demos are a bit mysterious to me - I just tried to leave them unchanged for the most part. 2 of 7 of them are actually still just broken, and I don't know how to fix them.

The behavior of the rest is of debatable value IMO. 

In this case, now that there can be automated tests that (hopefully) also demonstrate the behavior/usage of this code, I would actually argue for deleting these demos wholesale - but I will defer that to your and @lawhead 's judgement on this. 

At the very least though, I don't think it's worth spending time cleaning them up only to leave them in their current non-informative state; if it is worth having demos in addition to tests, then perhaps we should delete these and add a ticket to create some new meaningful ones?

Reply:
For now, I'm just following the approach of "preserve the demos, remove obvious crud and don't reformat" - and I'll add a ticket to discuss possibly updating or removing them later.